profiles/models.py

```diff
@@ -99,6 +100,25 @@ class Profile(models.Model):
     mailing_address = models.TextField(blank=True, null=True)
     date_joined_micromasters = models.DateTimeField(blank=True, null=True, auto_now_add=True)
     linkedin = JSONField(blank=True, null=True)
+    student_id = models.IntegerField(blank=True, null=True, unique=True)
+
+    @transaction.atomic
+    def save(self, *args, **kwargs):
+        """need to handle setting the student_id number"""
+        if self.id is None or self.student_id is None:
+            max_id = Profile.objects.aggregate(Max('student_id'))
+            if max_id['student_id__max'] is None:
+                max_id = 1
+            else:
+                max_id = max_id['student_id__max']
+            while Profile.objects.filter(student_id=max_id).exists():
+                max_id += 1
```

Comment:
@noisecapella I also think the right solution is not to let an exception be raised, but, if we really want to rely only on the DB transaction, we should have the field be a sequence (if we do not care about potential missing numbers).